pkg/primitives/vm/utils.go

```diff
@@ -121,11 +124,22 @@ func (p *Manager) prepContainer(
 		ReadOnly:        false,
 		PersistedVolume: volume.Path,
 	})
-
 	if err != nil {
 		return errors.Wrapf(err, "failed to mount flist: %s", wl.ID.String())
 	}
 
+	// clean up host keys
+	if isFirstBoot {
+		files, err := filepath.Glob(filepath.Join(mnt, "etc", "ssh", "ssh_host_*"))
+		if err != nil {
+			log.Fatal().Err(err)
```

Comment:
Are you aware that `Fatal` will cause the entire process to panic and crash? any call to `log.Fatal()` will immediately exist the process after logging the error because there is no way you can continue after a **fatal** error.

This error imho is completely ignorable, it's not an error that should be even logged as an error because this does not affect the operation of the system. Instead I would log this as a debug message and them move on

Also according to docs of `filepath.Glob`

> Glob ignores file system errors such as I/O errors reading directories. The only possible returned error is ErrBadPattern, when pattern is malformed.

Which means this will never return an error unless your `*` is wrong which is not the case here. (if files or path doesn't exist it won't mind)

```suggestion
			log.Debug().Err(err).Msg("failed to list ssh host keys for a vm image")
```

one last thing, A log with No `Msg()` call will not print anything. The only way for this error to print is to terminate it with a Message like in the suggestion above.